guides/getting-started.md

```diff
@@ -0,0 +1,1138 @@
+Use the `ec2_subnet` method to create an [AWS::EC2::Subnet][cf-subnet]
+resource and give it a CIDR block of 10.10.11.0/24. Here's the Convection code
+to do that.
+
+```ruby
+ec2_subnet 'PublicSubnet' do
+  network '10.10.11.0/24'
+  tag 'Name', "#{stack.cloud}-#{stack.name}-public"
+  vpc fn_ref('DemoVPC')
+end
+```
+
+Looking at the documentation for the [AWS::EC2::Subnet][cf-subnet]
+resource, the major difference between a public and private subnet is
+that Amazon assigns IP addresses to public subnets. Making a subnet public is
+a matter of setting the "MapPublicIpOnLaunch" property to `true`.
+
+We can look through the Convection code for the [ec2_subnet][ec2-subnet]
+method to figure out what the "MapPublicIpOnLaunch" property is named in
+Convection. It looks like the property's not there. Fortunately, we can extend
+Convection at runtime to include the property. This is a useful trick, since it
+lets us take advantage of new functionality in Amazon resources that might not
+yet be available through Convection.
+
+Update your Cloudfile to look like the one below.
+
+```ruby
+require 'convection'
+
+class Model::Template::Resource::EC2Subnet
+  property :public_ips, 'MapPublicIpOnLaunch', :default => false
+end
+
+region 'us-east-1'
+name 'convection-demo'
+
+vpc = template do
+  description 'VPC with Public and Private Subnets (NAT)'
+
+  ec2_vpc 'DemoVPC' do
+    network '10.10.10.0/23'
+    tag 'Name', "#{stack.cloud}-#{stack.name}"
+  end
+
+  ec2_subnet 'PrivateSubnet' do
+    network '10.10.10.0/24'
+    tag 'Name', "#{stack.cloud}-#{stack.name}-private"
+    vpc fn_ref('DemoVPC')
+  end
+
+  ec2_subnet 'PublicSubnet' do
+    network '10.10.11.0/24'
+    tag 'Name', "#{stack.cloud}-#{stack.name}-public"
+    vpc fn_ref('DemoVPC')
+    public_ips true
+  end
+end
+
+stack 'vpc', vpc
+```
+
+This adds a new property named "public_ips" to the `Mdoel::Template::Resource::EC2Subnet`
+class in Convection. That property sets the "MapPublicIpOnLaunch" property in
+CloudFormation. We'll give it a default value of `false` so we don't break our
+private subnet when we create this new public subnet.
+
+As before, we'll diff our template before converging it to make sure it's doing
+what we expect.
+
+```text
+$> convection diff
+
+compare  Compare local state of stack vpc (convection-demo-vpc) with remote template
+ delete  .Resources.PrivateSubnet.Properties.MapPublicIpOnLaunch
+ create  .Resources.PublicSubnet.Type: AWS::EC2::Subnet
+ create  .Resources.PublicSubnet.Properties.VpcId.Ref: DemoVPC
+ create  .Resources.PublicSubnet.Properties.CidrBlock: 10.10.11.0/24
+ create  .Resources.PublicSubnet.Properties.MapPublicIpOnLaunch: true
+ create  .Resources.PublicSubnet.Properties.Tags.0.Key: Name
+ create  .Resources.PublicSubnet.Properties.Tags.0.Value: convection-demo-vpc-public
+```
+
+Convection says it will delete the "MapPublicIpOnLaunch" from our private
+subnet. That's fine, because the subnet is private. The public subnet will be
```

Comment:
The sentence on 401-402 would benefit from a little bit extra about why `MapPublicIpOnLaunch` is going to be deleted from the private subnet.